src/lib/work_partitioner/work_partitioner.mli

```diff
@@ -0,0 +1,15 @@
+open Core_kernel
+module Snark_worker_shared = Snark_worker_shared
+
+type t
+
+val create : reassignment_timeout:Time.Span.t -> logger:Logger.t -> t
+
+type work_from_selector =
+  Snark_work_lib.Spec.Single.t One_or_two.t option Lazy.t
+
+val request_partitioned_work :
```

Comment:
Add explanation that error is propagated from splitting (you know, extracting witnesses or whatever that is called)